command/debug.go

```diff
@@ -255,26 +345,58 @@ func (c *DebugCommand) Run(args []string) int {
 
 	// Capture static information
 	c.UI.Info("==> Capturing static information...")
-	if err := c.captureStaticTargets(); err != nil {
-		c.UI.Error(fmt.Sprintf("Error capturing static information: %s", err))
-		return 2
-	}
 
+	var wg sync.WaitGroup
+	errs := make([]error, len(collectors))
+	for i := range collectors {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			errs[i] = collectors[i].captureStaticTargets()
+		}(i)
+	}
+	wg.Wait()
+	for i, err := range errs {
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error capturing static information for address %s: %s", c.flagAddresses[i], err))
+			return 2
+		}
+	}
 	c.UI.Output("")
 
 	// Capture polling information
 	c.UI.Info("==> Capturing dynamic information...")
-	if err := c.capturePollingTargets(); err != nil {
-		c.UI.Error(fmt.Sprintf("Error capturing dynamic information: %s", err))
-		return 2
+
+	for i := range collectors {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			errs[i] = collectors[i].capturePollingTargets()
+		}(i)
 	}
+	wg.Wait()
+	for i, err := range errs {
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error capturing static information for address %s: %s", c.flagAddresses[i], err))
```

Comment:
polling info rather than static? error message is the same as above

Reply:
Fixed, ty